x/tokenfactory/client/cli/cli_test.go

```diff
@@ -0,0 +1,489 @@
+				"cosmos1s0he0z3g92zwsxdj83h0ky9w463sx7gq9mqtgn",
+				time.Date(2020, 1, 1, 12, 00, 00, 000, time.UTC),
+				nil,
+			),
+		},
+		nil, nil, nil, nil, nil,
+	)
+
+	subspacesDataBz, err := cfg.Codec.MarshalJSON(subspacesGenesis)
+	s.Require().NoError(err)
+	genesisState[subspacestypes.ModuleName] = subspacesDataBz
+
+	tokenfactoryGenesis := &types.GenesisState{
+		Params: types.DefaultParams(),
+		FactoryDenoms: []types.GenesisDenom{
+			{
+				Denom: "factory/cosmos1cyjzgj9j7d2gdqk78pa0fgvfnlzradat97aek9/uminttoken",
+				AuthorityMetadata: types.DenomAuthorityMetadata{
+					Admin: "cosmos1s0he0z3g92zwsxdj83h0ky9w463sx7gq9mqtgn",
+				},
+			},
+			{
+				Denom: "factory/cosmos1cyjzgj9j7d2gdqk78pa0fgvfnlzradat97aek9/utesttoken",
+				AuthorityMetadata: types.DenomAuthorityMetadata{
+					Admin: "cosmos1cyjzgj9j7d2gdqk78pa0fgvfnlzradat97aek9",
+				},
+			},
+		},
+	}
+
+	tokenfactoryDataBz, err := cfg.Codec.MarshalJSON(tokenfactoryGenesis)
+	s.Require().NoError(err)
+	genesisState[types.ModuleName] = tokenfactoryDataBz
+
+	cfg.GenesisState = genesisState
+
+	s.cfg = cfg
+	s.network, err = network.New(s.T(), s.T().TempDir(), cfg)
+	s.Require().NoError(err)
+
+	_, err = s.network.WaitForHeight(1)
+	s.Require().NoError(err)
+}
+
+func (s *IntegrationTestSuite) TearDownSuite() {
+	s.T().Log("tearing down integration test suite")
+	s.network.Cleanup()
+}
+
+// --------------------------------------------------------------------------------------------------------------------
+
+func (s *IntegrationTestSuite) TestCmdQueryParams() {
+	val := s.network.Validators[0]
+	testCases := []struct {
+		name        string
+		args        []string
+		shouldErr   bool
+		expResponse types.QueryParamsResponse
+	}{
+		{
+			name: "params is returned properly",
+			args: []string{
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			},
+			shouldErr: false,
+			expResponse: types.QueryParamsResponse{
+				Params: types.DefaultParams(),
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		s.Run(tc.name, func() {
+			cmd := cli.GetCmdQueryParams()
+			clientCtx := val.ClientCtx
+			out, err := clitestutil.ExecTestCLICmd(clientCtx, cmd, tc.args)
+
+			if tc.shouldErr {
+				s.Require().Error(err)
+			} else {
+				s.Require().NoError(err)
+
+				var response types.QueryParamsResponse
+				s.Require().NoError(clientCtx.Codec.UnmarshalJSON(out.Bytes(), &response), out.String())
+				s.Require().Equal(tc.expResponse.Params, response.Params)
+			}
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestCmdQuerySubspaceDenoms() {
+	val := s.network.Validators[0]
+	testCases := []struct {
+		name        string
+		args        []string
+		shouldErr   bool
+		expResponse types.QuerySubspaceDenomsResponse
+	}{
+		{
+			name: "invalid subspace id returns error",
+			args: []string{
+				"X",
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			},
+			shouldErr: true,
+		},
+		{
+			name: "denoms are returned properly returns error",
```

Comment:
This should be removed
```suggestion
			name: "denoms are returned properly",
```